src/main/kotlin/org/jmailen/gradle/kotlinter/tasks/EditorConfigUtils.kt

```diff
@@ -0,0 +1,31 @@
+package org.jmailen.gradle.kotlinter.tasks
+
+import org.gradle.api.file.ProjectLayout
+import java.io.File
+
+internal fun ProjectLayout.findApplicableEditorConfigFiles(): Sequence<File> {
+    val projectEditorConfig = projectDirectory.file(".editorconfig").asFile
+
+    return generateSequence(seed = projectEditorConfig) { editorconfig ->
+        println("Invoked with $editorconfig")
+        if (editorconfig.isRootEditorConfig) {
+            null
+        } else {
+            editorconfig.parentFile?.parentFile?.resolve(".editorconfig")
+        }
+    }
+}
+
+private val File.isRootEditorConfig: Boolean
+    get() {
+        if (!isFile || !canRead()) return false
+
+        return useLines { lines ->
+            lines.any { line -> line.matches(editorConfigRootRegex) }
+        }
+    }
+
+/**
+ * According to https://editorconfig.org/ root-most EditorConfig file contains line with `root=true`
+ */
+private val editorConfigRootRegex = "^root\\s?=\\s?true".toRegex()
```

Comment:
Since we don't tell ktlint what editorconfig file to use, the tricky bit is that we're trying to match what they'll pick up.
What would be best is if they supplied a getter for editorconfig files as part of their "api". For example `getConfigurations()`. But it probably wouldn't have made sense to them why they need it unless they were familiar with gradle plugins.

Reply:
Based on your comment in `ktlint` issue, I filed an issue: https://github.com/pinterest/ktlint/issues/1446

Regarding the proposed changes: I initially tried to use the same library as `ktlint` does (https://github.com/ec4j/ec4j) but I wasn't able to find a way to identify file locations (the library returns already loaded properties :/)
I decided to do a similar thing as `ktlint-gradle` does as it felt like a low risk decision. Assuming my implementation will falsely include an unrelated `.editorconfig` files - the penalty is `LintTask` will be invoked once, instead of being up-to-date (this assumes nobody changes `.editorconfig` files on a daily basis 😛)

My biggest concern here was breaking Remote Cache, when the cache is relocated we'll point files outside a `rootProject`. I initially assumed this _should_ work as we're referencing these files by relative location, but now I started wondering if people using Remote Cache won't start seeing cache misses more often 🤔 
Maybe I should clamp down the traversal to `rootProject`? It would be less _correct_ as Ktlint will still access files up to very root `/.editorconfig`, but it should be _safer_ in terms of relocating the cache and at the same time will cover most of use-cases

Please do not feel an urge to merge this PR. We have a workaround I shared in the linked issue to add inputs manually, so there is not much of a pressure :)

Reply:
Good concerns although I think you are right that it would be rare to find an .editorconfig file which is changed.

At one point I was almost of the mind that we should consider only an .editorconfig directly in the root of the project, but I know as soon as we do that, someone will have a use case :). I assume ktlint does the right thing and actually enforces, say, indents differently depending on which submodule it is in and the closest .editorconfig file 🤔.